### ISSUE TITLE
Crate fails to build on platforms outside x86 / x86_64

### ISSUE THREAD

**Issue by Lokathor** [tkaitchuck/aHash#37]
You've set up `shuffle` in `operations.rs` to use `_mm_shuffle_epi8` or not based on the result of the `cfg!` macro. This is incorrect because on non-{x86 | x86_64} builds this will attempt to compile in a call to `_mm_shuffle_epi8` and then just never take that code path. This is enough to make the build fail because of name resolution failure. That function doesn't exist at all outside x86 / x86_64.

Here is a corrected version of `shuffle`:

```rust
#[inline(always)]
pub(crate) fn shuffle(a: u128) -> u128 {
    #[cfg(all(target_feature = "ssse3", not(miri)))]
    {
        use core::mem::transmute;
        #[cfg(target_arch = "x86")]
        use core::arch::x86::*;
        #[cfg(target_arch = "x86_64")]
        use core::arch::x86_64::*;
        unsafe {
            transmute(_mm_shuffle_epi8(transmute(a), transmute(SHUFFLE_MASK)))
        }
    }
    #[cfg(not(all(target_feature = "ssse3", not(miri))))]
    {
        a.swap_bytes()
    }
}
```

**Comment by tkaitchuck** [comment]
Fixed in 0.4.1 https://github.com/tkaitchuck/aHash/commit/465c9aae391f5cc840ed4c5da3f9fd6b3f5fdb1a

**Comment by tkaitchuck** [comment]
Added CI for Arm7.
